src/amo/components/Collection/index.js

```diff
@@ -114,6 +110,10 @@ export class CollectionBase extends React.Component<Props> {
     // When switching into edit mode, refresh the collection add-ons.
     if (this.props.editing && !nextProps) {
       addonsPageChanged = true;
+    } else if (loading && !this.props.editing) {
```

Comment:
Thanks @rebmullin. This fixes the problem, but I'm a bit confused about it. If `loading` is true, then why do we need to not `return` when we're in edit mode? It seems like if `loading` is true then we're waiting for a fetch action to return, and that should be true whether we're in edit mode or not. For some reason, in edit mode, when we refresh the page, `loading` is true, but yet a fetch is never completed which updates `loading` to false.

That seems like a bug, and I'm concerned that this fix is just a band-aid that will hide that other problem. Can you look into it a bit more to see if you can figure out the source of the issue I just described?

Reply:
thanks @bobsilverberg. I'll take another look